GameLib/AbstractPlayer.h

```diff
@@ -1,26 +1,33 @@
 #pragma once
+
 #include "Barrier.h"
-#include "Position.h"
+#include "ConsoleUtils.h"
 #include "Direction.h"
+#include "Position.h"
 #include <cstddef>
 #include <vector>
 
+
 class AbstractPlayer
 {
 public:
-    AbstractPlayer(const Position& startPos);
+    explicit AbstractPlayer(Coordinate startCoords, const std::vector<ConsoleUtils::Colors> &player_color);
     virtual ~AbstractPlayer() = 0;
     virtual void playerTurn() = 0; // Turn of player (console dialog or automatic for bot player)
 
-    Position getPosition();
+    Coordinate getCoordinate();
+    std::string toString();
+    bool hasVerticalBarrier(Coordinate &coordinate);
+    bool hasHorizontalBarrier(Coordinate &coordinate);
```

Comment:
ich dachte gamefield bzw der graph haben die infos über die barriers schon. und barriers gehören meiner meinung in gamefield.

Reply:
kann man glaub raus machen oder? mit der korrekt implementierten barrier.

Reply:
Ja sollte man können